### PR TITLE
adding SAM32 board

### PR DESCRIPTION
**Pull request by maholli**
SAMD51 running circuitpython talking to an ESP32 running micropython.

Full board details at: [https://github.com/maholli/SAM32](https://github.com/maholli/SAM32)

### CODE REVIEW

File: ports/atmel-samd/boards/sam32/mpconfigboard.mk [hunk 1]
```diff
@@ -0,0 +1,12 @@
+LD_FILE = boards/samd51x20-bootloader.ld
+USB_VID = 0x239A
+USB_PID = 0x8021
```
Comment (sommersoft) [inline ports/atmel-samd/boards/sam32/mpconfigboard.mk]: I think Adafruit might appreciate you using a non-Adafruit VID & PID.

File: ports/atmel-samd/boards/sam32/pins.c [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    { MP_ROM_QSTR(MP_QSTR_RTS),  MP_ROM_PTR(&pin_PB11)  },
+    { MP_ROM_QSTR(MP_QSTR_DTR),  MP_ROM_PTR(&pin_PB10)  },
+
+    { MP_ROM_QSTR(MP_QSTR_LED),       MP_ROM_PTR(&pin_PA27)  },
+    { MP_ROM_QSTR(MP_QSTR_NEOPIXEL),  MP_ROM_PTR(&pin_PA15)  },
+    
+    { MP_ROM_QSTR(MP_QSTR_UART), MP_ROM_PTR(&board_uart_obj) },
+    { MP_ROM_QSTR(MP_QSTR_I2C),  MP_ROM_PTR(&board_i2c_obj)  },
+    { MP_ROM_QSTR(MP_QSTR_SPI),  MP_ROM_PTR(&board_spi_obj)  },
+    { MP_ROM_QSTR(MP_QSTR_VBAT_MES),   MP_ROM_PTR(&pin_PA06) },
```
Comment (tannewt) [inline ports/atmel-samd/boards/sam32/pins.c]: I think we're standardizing on "BATTERY" for the voltage monitoring pin's name.

File: .travis.yml [hunk 3]
```diff
@@ -25,7 +25,7 @@ env:
   - TRAVIS_BOARDS="metro_m0_express metro_m4_express pirkey_m0 trellis_m4_express trinket_m0 sparkfun_lumidrive sparkfun_redboard_turbo" TRAVIS_SDK=arm
   - TRAVIS_BOARDS="feather_radiofruit_zigbee gemma_m0 hallowing_m0_express itsybitsy_m0_express itsybitsy_m4_express meowmeow" TRAVIS_SDK=arm
   - TRAVIS_BOARDS="feather_m0_express_crickit feather_m0_rfm69 feather_m0_rfm9x feather_m4_express arduino_zero arduino_mkr1300 arduino_mkrzero" TRAVIS_SDK=arm
-  - TRAVIS_BOARDS="circuitplayground_express_crickit feather_m0_adalogger feather_m0_basic feather_m0_express catwan_usbstick pyportal sparkfun_samd21_mini sparkfun_samd21_dev" TRAVIS_SDK=arm
+  - TRAVIS_BOARDS="circuitplayground_express_crickit feather_m0_adalogger feather_m0_basic feather_m0_express catwan_usbstick pyportal sparkfun_samd21_mini sparkfun_samd21_dev sam32" TRAVIS_SDK=arm
```
Comment (tannewt) [inline .travis.yml]: I would put this on a different list. This last one takes the longest to run. How about the third on instead? It looks like it's the fastest: https://travis-ci.com/adafruit/circuitpython

Comment (sommersoft) [reply .travis.yml]: Oops! I didn't even think about the runtimes when I suggested the last line.

**Pull request merged**